include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,648 @@
+    decoded_view(
+        pct_decode_opts const& opt = {}) noexcept
+        : p_(nullptr)
+        , n_(0)
+        , dn_(0)
+        , plus_to_space_(opt.plus_to_space)
+    {}
+
+    /** Constructor
+    */
+    BOOST_URL_DECL
+    decoded_view(
+        string_view str,
+        pct_decode_opts opt = {}) noexcept;
+
+    /** Returns an iterator to the beginning
+    */
+    const_iterator
+    begin() const noexcept;
+
+    /** Returns an iterator to the end
+    */
+    const_iterator
+    end() const noexcept;
+
+    /** Accesses the first decoded character
+
+        @note The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+    */
+    const_reference
+    front() const;
+
+    /** Accesses the last decoded character
+
+        @note The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+    */
+    const_reference
+    back() const;
+
+    /** Returns the encoded string view
+
+        This function returns a string_view of
+        the underlying encoded string.
+
+        @return The encoded string view.
+    */
+    string_view
+    encoded() const
+    {
+        return {p_, n_};
+    }
+
+    /** Returns the number of char elements in the string
+    */
+    size_type
+    size() const noexcept
+    {
+        return dn_;
+    }
+
+    /** Returns whether the string is empty
+    */
+    bool
+    empty() const noexcept
+    {
+        return n_ == 0;
+    }
+
+    /** Copies a decoded substring to another character string
+
+        Copies a substring to the character array pointed to by
+        the destination char string, where `rcount` is the smaller
+        of `count` and `size() - pos`.
+
+        @par Exception Safety
+        Strong guarantee.
+        Exceptions thrown on invalid positions.
+
+        @throw std::out_of_range If `pos > size()`
+
+        @param dest pointer to the destination character string
+        @param count requested substring length
+        @param pos position of the first character
+
+        @return Number of characters copied
+    */
+    BOOST_URL_DECL
+    size_type
+    copy(char* dest,
+         size_type count,
+         size_type pos = 0) const;
+
+    /** Return true if the corresponding comparison holds
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.data(), rlen)`.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(string_view other) const noexcept;
+
+    /// @copydoc compare()
+    int
+    compare(char const* other) const noexcept
```

Comment:
But why do we need this overload if we have `string_view`? Does it not compile otherwise?